File: 2022/Python/Day 2/AoC-Day2-Part 1.py
def D2P1(Input):
    input_file = open(Input,"r")
    InputList = input_file.read().splitlines()

    score = {
      'A':1,
      'B':2,
      'C':3,
      'X':1,
      'Y':2,
      'Z':3
    }
      # Rock A 1 vs Paper Y 2     P2 Win Diff:-1
      # Rock A 1 vs Scissor Z 3   P1 Win Diff:-2

      #Scissor C 3 vs Rock X 1 .  P2 Win Diff:2
      #Scissor C 3 vs Paper Y 2   P1 Win Diff:1

      #Paper B 2 vs Rock X 1      P1 Win Diff:1
      #Paper B 2 vs Scissor Z 3   P2 Win Diff:-1

    total_score = 0
    for line in InputList:
      opponent_score = score[line[0]]
      my_score = score[line[2]]
      if opponent_score == my_score:
        total_score += 3
      elif (opponent_score - my_score) in [-1,2]:
        total_score += 6
      total_score += my_score

    return total_score

assert D2P1("D2 Test.txt") == 15, "Function does not pass example given"

if __name__ == "__main__":
    print(D2P1("D2 Input.txt"))

File: 2022/JavaScript/Day 2/AoC-Day2-Part1.js
const { readFileSync } = require('fs')

const handleScoring = (line) => {

  const score = {
    A:1,
    B:2,
    C:3,
    X:1,
    Y:2,
    Z:3
  }

  let picks = line.split(" ")
  let opponentScore = score[picks[0]]
  let myScore = score[picks[1]]
  if (opponentScore === myScore){
    return 3 + myScore
  } else if (opponentScore-myScore === -1 || opponentScore-myScore === 2){
    return 6 + myScore
  } else {
    return myScore
  }
}

const D2P1 = (input) => {
  const contents = readFileSync(input,'utf-8')

  let arr = contents.split(/\n/)

  output = arr.map(line => handleScoring(line)).reduce( (acc,cur) => acc + cur,0)

  return output
}

const assert= (condition) => {
  if (!condition){
    throw Error('Function does match example given')
  }
}

assert(D2P1('./D2 Test.txt')===15)

console.log(D2P1('./D2 Input.txt'))